bin/github-backup

```diff
@@ -382,24 +389,71 @@ def get_github_repo_url(args, repository):
     return repo_url
 
 
+def _update_githubapp_token(current_token):
+    global _token_cache
+    token_path = os.path.expanduser('~/.gh-token')
+
+    if os.path.exists(token_path):
+        last_modified = os.path.getmtime(token_path)
+
+        if last_modified > _token_cache['last_modified']:
+            with open(token_path, "r") as f:
+                new_token = f.read().strip()
+            _token_cache['token'] = new_token
+            _token_cache['last_modified'] = last_modified
+
+        if current_token != _token_cache['token']:
+            return _token_cache['token']
+    else:
+        log_error('No token found in ~/.gh-token')
+
+    return current_token
+
+
 def retrieve_data(args, template, query_args=None, single_request=False):
     auth = get_auth(args)
+    
     query_args = get_query_args(query_args)
     per_page = 100
     page = 0
     data = []
 
+    if args.token:
+        auth = _update_githubapp_token(auth)
+ 
     while True:
         page = page + 1
-        request = _construct_request(per_page, page, query_args, template, auth)  # noqa
+        request = _construct_request(per_page, page, query_args, template)
+        if args.token and auth is not None:
+            request.add_header('Authorization', 'Bearer ' + auth)  # noqa
+        elif args.username and auth is not None:
+            print("Have auth initialized")
+            log_info("Auth is initialized")
+            log_info("Adding to header")
+            request.add_header('Authorization', 'Basic '.encode('ascii') + auth)
         r, errors = _get_response(request, auth, template)
 
         status_code = int(r.getcode())
 
         if status_code != 200:
-            template = 'API request returned HTTP {0}: {1}'
-            errors.append(template.format(status_code, r.reason))
-            log_error(errors)
+            err_template = 'API request returned HTTP {0}: {1}'
+            errors.append(err_template.format(status_code, r.reason))
+            if status_code == 401:
```

Comment:
By calling get_auth() on every call, which looks for updates to the token file, we could dispense with this special handling of 401s. We could let the 401s fail since they would not be caused by expired tokens.